src/oom/oomd-util.c

```diff
@@ -278,6 +284,25 @@ int oomd_cgroup_context_acquire(const char *path, OomdCGroupContext **ret) {
         if (r < 0)
                 return log_debug_errno(r, "Error parsing memory pressure from %s: %m", p);
 
+        r = cg_get_owner(SYSTEMD_CGROUP_CONTROLLER, path, &uid, &gid);
+        if (r < 0)
+                log_debug_errno(r, "Failed to get owner/group from %s: %m", path);
+        else if (uid == 0 && gid == 0) {
+                /* Ignore most errors when reading the xattr since it is usually unset and cgroup xattrs are only used
+                 * as an optional feature of systemd-oomd (and the system might not even support them). */
+                r = cg_get_xattr_malloc(SYSTEMD_CGROUP_CONTROLLER, path, "user.oomd_avoid", &avoid_val);
+                if (r >= 0 && streq(avoid_val, "1"))
```

Comment:
use parse_boolean() here, it think that wouldn't hurt and is easier on the eyes... maybe even add a new helper cg_get_xattr_boolean() that wraps all this. i.e. do what getenv_boolean() is for getenv() but for xattrs